### PR TITLE
Add unpack_forward_output handlers and overrides for training

### PR DESCRIPTION
**Pull request by agobeljicTT**
> [!NOTE]
> This PR complements [tt-xla/pull/4549](https://github.com/tenstorrent/tt-xla/pull/4549) which adds the triage skill that produced these handlers/overrides and updates the test YAMLs to reflect the new failure stages.

### Ticket
Refs: [tenstorrent/tt-xla#4293](https://github.com/tenstorrent/tt-xla/issues/4293) (kept open — this is only the first pattern of FE failures being triaged).

### Problem description
A number of training tests in tt-xla were stuck at `FAILED_FE_COMPILATION` with reason:

> `tt-forge-models doesn't implement unpack_forward_output for this model.`

The default `unpack_forward_output` (`base.py` -> `training_utils.py`) is keyed by output class name. When the model returns a class not in the registry (a bare `list`/`tuple`/`dict`, or a HuggingFace `ModelOutput` subclass not yet registered), the call raises and training cannot proceed.

### What's changed
Two complementary fixes — one or the other is picked per model based on whether the output has a stable class name to key on:

**Registry entries in `training_utils.py`** (preferred for HuggingFace `ModelOutput` dataclasses):
- DPR: `DPRContextEncoderOutput`, `DPRQuestionEncoderOutput` -> `pooler_output`
- Detection: `DetrObjectDetectionOutput`, `DFineObjectDetectionOutput`, `YolosObjectDetectionOutput`, `RTDetrObjectDetectionOutput` -> `logits`
- Segmentation: `DetrSegmentationOutput`, `SamImageSegmentationOutput` -> `pred_masks`; `SemanticSegmenterOutput` -> `logits`; `Swinv2MaskedImageModelingOutput` -> `reconstruction`
- Classification: `SwinImageClassifierOutput`, `Swinv2ImageClassifierOutput` -> `logits`
- QA: `QuestionAnsweringModelOutput` -> `start_logits`
- Vision-language: `Qwen2VLCausalLMOutputWithPast` -> `logits`
- Custom handler for `MgpstrModelOutput` removed from registry (the `logits` attribute is a 3-tuple, not a tensor — registry would have raised); replaced with a per-loader override that sums across all three char/bpe/wp heads so all participate in backward.

**Per-loader `unpack_forward_output` overrides** (required for bare `list`/`tuple`/`dict` outputs, or outputs whose structure warrants selecting specific tensors):
- `bge_m3/pytorch/loader.py` — bare dict; selects `dense_vecs`.
- `efficientdet/pytorch/loader.py` — tuple of two 5-scale lists (cls + box); sums all 10 feature maps.
- `fpn/pytorch/loader.py` — bare tuple of 4 feature maps; sums all four.
- `maskformer_swin_b/pytorch/loader.py` — selects `class_queries_logits` and `masks_queries_logits`; sums both heads.
- `mgp_str_base/pytorch/loader.py` — `MgpstrModelOutput.logits` is a 3-tuple; sums all three head tensors.
- `retinanet/pytorch/loader.py` — list of 10 cls/reg feature maps; sums all.
- `siglip/image_text_similarity/pytorch/loader.py` — returns `logits_per_text` (mirrors existing `CLIPOutput` entry).
- `yolov3/pytorch/loader.py` — list of 3 per-scale detection tensors; sums all.
- `yolov6/pytorch/loader.py` — `[predictions, featmaps]`; selects `predictions`.
- `yoloworld/pytorch/loader.py` — nested detection-head output; recursively extracts and sums all tensors.

Each override docstring records the output structure, what is being selected as the gradient sink and why, and why a registry entry was not sufficient.

### Checklist
- [x] New/Existing tests provide coverage for changes (existing training tests in tt-xla — they were the ones gating on this failure, and they now advance past FE compilation; outcomes recorded in the companion tt-xla PR's YAML updates).

### CODE REVIEW

File: bge_m3/pytorch/loader.py [hunk 1]
```diff
@@ -102,3 +102,20 @@ def load_config(self):
...
+
+        What is selected and why: returning ``dense_vecs`` -- the pooled
+        sentence embedding through which the contrastive retrieval loss
+        backpropagates through the entire XLM-Roberta encoder. The sparse and
+        ColBERT heads are dropped: each is a single linear projection on top
+        of the same encoder, so omitting them only skips two small head layers
+        while still exercising the bulk of the parameters.
+
+        Why a registry entry was not sufficient: the forward returns a bare
+        ``dict``, which has no class name the registry can key on.
```
Comment (ndrakulicTT) [inline bge_m3/pytorch/loader.py]: We probably dont need this comment, why it was not in the registry, not sure what is your opinion on this?

**Review by ndrakulicTT — APPROVED** [review]
This fixes seams great. LGTM

File: bge_m3/pytorch/loader.py (hunk 1 above)
Comment (agobeljicTT) [reply bge_m3/pytorch/loader.py]: perhaps we don't need it, but then again it's a patch, worths noting why it is present... idk tbh

File: fpn/pytorch/loader.py [hunk 2]
```diff
@@ -129,3 +129,19 @@ def load_inputs(self, batch_size=1, dtype_override=None):
...
+        What is selected and why: flatten and concatenate all four maps. They
+        are the entirety of the FPN's output and the gradient sink for any
+        downstream detector; backward through the concatenation trains the
+        FPN's lateral 1x1 + smooth 3x3 conv stack. The flattened concatenation
+        is a 1D tensor, not the per-anchor / per-cell layout the upstream loss
+        consumes -- it drives backward through the full graph correctly, but
+        downstream numeric comparison against the torch loss output should
+        account for this layout difference.
+        """
+        return torch.cat([t.flatten() for t in fwd_output])
```
Comment (ndrakulicTT) [inline fpn/pytorch/loader.py]: Maybe is better to do sum, something like `torch.sum([t.sum() for t in fwd_output])`, this will produce the shape that is not ideal for our tt hardwere as we are tiling everything to 32x32

Comment (ndrakulicTT) [reply fpn/pytorch/loader.py]: same comment for other places

**Review by ndrakulicTT — APPROVED** [review]

**Review by ppadjinTT — APPROVED** [review]
lgtm
